app/lib/shared/handler_helpers.dart

```diff
@@ -63,6 +63,7 @@ shelf.Handler wrapHandler(
   // Handlers wrap other handlers, and they are called in the revers order of
   // their wrapping. Read this list from the bottom and go up to get the real
   // execution order.
+  handler = _redirectLoopDetectorWrapper(logger, handler);
   handler = _cspHeaderWrapper(handler);
   handler = _userAuthWrapper(handler);
   handler =
```

Comment:
@isoos 
unrelated nit: But should we perhaps use [shelf.Pipeline](https://pub.dev/documentation/shelf/latest/shelf/Pipeline-class.html) here?

Reply:
Yeah, we are at the point where it would make sense :)